source/System.cpp

```diff
@@ -310,7 +310,35 @@ void System::Load(const DataNode &node, Set<Planet> &planets)
 		else if(key == "object")
 			LoadObject(child, planets);
 		else if(key == "arrival")
-			extraArrivalDistance = child.Value(valueIndex);
+		{
+			bool loadHyperDistance = true;
+			bool loadJumpDistance = true;
+			// Check if we load arrival distances specifically for hyperdrives
+			// or jump drives only.
+			if(child.Size() >= 3)
+			{
+				loadHyperDistance = false;
+				loadJumpDistance = false;
+				for(int i = 2; i < child.Size(); i++)
+				{
+					if(child.Token(i) == "link")
+						loadHyperDistance = true;
+					else if(child.Token(i) == "jump")
+						loadJumpDistance = true;
+					else
+						child.PrintTrace("Warning: Unknown arrival type \"" + child.Token(i) + "\" for arrival distance:");
+				}
+			}
+			if(loadHyperDistance)
+				extraHyperArrivalDistance = child.Value(valueIndex);
+			// We only load additional jump distances if they are above 0.
+			// Jump drives use a circle around the target for targeting,
+			// so a value below 0 doesn't have a meaning in the same way
+			// as that a hyper-jump-distance negative values would be beyond
+			// the target.
```

Comment:
I imagine the behavior of a negative value is just the same as the same magnitude positive value, so why explicitly prevent the use of negative values if the end behavior doesn't actually cause any issues? Seems to me like it'd make more sense to just allow both.

Reply:
The behaviour for Hyperdrives is very different, in that a negative value will cause the ship to overshoot the target.

What I can do for Jump-drives is to indeed make negative and positive values act in the same way.

Reply:
Updated to allow negative values, but convert them to positive internally (so that the ring width calculation is not affected).